### DISCUSSION TITLE
How to switch to dev branch properly?

### DISCUSSION THREAD

**Discussion by ke1ne** [vladmandic/sdnext#2303] [Q&A] [answered]
Hi, I want to play with the dev version. But it resets back to main every time with --upgrade.
Please give some advise, how to do this properly. Thanks!

**Comment by brknsoul** [comment]
I'd just clone the dev branch in a new folder

`git clone -b dev https://github.com/vladmandic/automatic sdnext-dev`

Start `webui.bat` with adding `--models-dir MODELS_DIR` to the rest of your normal arguments.
Where MODELS_DIR points to your master install's models folder.
For example;
`webui.bat --models-dir \automatic\models`

**Comment by ghost** [comment]
If I install the dev branch, and the release comes, how do I switch to master in that folder I installed, so I can make the master git pull? Is that even a "clean" method to update or should I handle the dev branch always as a separate instance?

**Reply by vladmandic** [reply, marked as answer]
```
git checkout dev
git pull
```

```
git checkout master
git pull
```

**Reply by SAC020** [reply]
I applied the method above and now I see in the log

```
09:16:00-818262 INFO     Version: app=sd.next updated=2024-12-09 hash=a2f5a39a branch=master
                         url=https://github.com/vladmandic/automatic/tree/master ui=dev
09:16:00-819350 DEBUG    Branch mismatch: sdnext=master ui=dev
09:16:01-203500 DEBUG    Branch sync failed: sdnext=master ui=main
```


How do I fix it? (does it need fixing?)

**Comment by ke1ne** [comment]
Thank you, dudes, it works
